nni/nas/execution/common/integration_api.py

```diff
@@ -41,7 +45,11 @@ def send_trial(parameters: dict, placement_constraint=None) -> int:
     Send a new trial. Executed on tuner end.
     Return a ID that is the unique identifier for this trial.
     """
-    return get_advisor().send_trial(parameters, placement_constraint)
+    advisor = get_advisor()
+    while not advisor.initialized:
```

Comment:
Suggest putting this into `RetiariiAdvisor.send_trial`.

Reply:
updated